src/orc.cpp

```diff
@@ -444,37 +289,48 @@ die* enforce_odrv_for_die_list(die* base, std::vector<odrv_report>& results) {
 
     static TracyLockable(std::mutex, odrv_report_mutex);
     {
-    std::lock_guard<LockableBase(std::mutex)> lock(odrv_report_mutex);
-    results.push_back(report);
+        std::lock_guard<LockableBase(std::mutex)> lock(odrv_report_mutex);
+        results.push_back(report);
     }
 
     return dies.front();
 }
 
 /**************************************************************************************************/
 
-std::vector<odrv_report> orc_process(const std::vector<std::filesystem::path>& file_list) {
-    // First stage: process all the DIEs
+std::vector<odrv_report> orc_process(std::vector<std::filesystem::path>&& file_list) {
+    // First stage: (optional) dependency/dylib preprocessing
+    if (settings::instance()._dylib_scan_mode) {
```

Comment:
This is the main engine for ORC, and this new block adds the dylib scan mode as the first step (when asked for.)